src/preact/base-element.js

```diff
@@ -153,9 +153,28 @@ export class PreactBaseElement extends AMP.BaseElement {
       notify: () => this.mutateElement(() => {}),
     };
 
+    /** @private {?API_TYPE} */
+    this.apiWrapper_ = null;
+
     /** @private {{current: ?API_TYPE}} */
     this.ref_ = createRef();
 
+    /** @param {?API_TYPE|null} current */
+    this.refSetter_ = (current) => {
+      // The API shape **must** be consistent.
```

Comment:
Is this enforced somehow? How easy would it be to make this mistake?

Reply:
Update: It's now enforced at dev time, you'll receive errors if it happens.